arroyo/examples/transform_and_produce/produce_step.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+from collections import deque
+from concurrent.futures import Future
+
+import logging
+from typing import (
+    Callable,
+    Deque,
+    Mapping,
+    NamedTuple,
+    Optional,
+)
+from arroyo.backends.kafka.consumer import KafkaPayload, KafkaProducer
+from arroyo.processing.strategies.abstract import ProcessingStrategy as ProcessingStep
+from arroyo.types import Message, Partition, Position, Topic
+
+logger = logging.getLogger(__name__)
+
+
+class MessageToFuture(NamedTuple):
+    """
+    Map a submitted message to a Future returned by the Producer.
+
+    This is useful for being able to commit the latest offset back
+    to the original consumer.
+    """
+
+    message: Message[KafkaPayload]
+    future: Future[Message[KafkaPayload]]
+
+
+class ProduceStep(ProcessingStep[KafkaPayload]):
+    """
+    Simple step that produces a given message onto a topic.
```

Comment:
Agree it's not super complex but I don't think characterizing this as "simple" is particularly helpful to new users. I think we can try to give a more helpful description here.

```suggestion
    Produces a message onto a topic asynchronously and commits the offset of the original message once producing is complete.
```